frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp

```diff
@@ -127,6 +198,66 @@ void AcapController::fallbackKernel(const OperatorHandle &opHandle,
   current->fallbackKernelImpl(opHandle, stack);
 }
 
+at::Tensor AcapController::convolutionKernel(
+    const at::Tensor &input, const at::Tensor &weight,
+    const c10::optional<at::Tensor> &bias, const at::IntArrayRef stride,
+    const at::IntArrayRef padding, const at::IntArrayRef dilation,
+    const bool transposed, const at::IntArrayRef output_padding,
+    const int64_t groups) {
+  static c10::OperatorName opName{"aten::convolution", ""};
+  auto &dispatcher = c10::Dispatcher::singleton();
+  auto opHandle = dispatcher.findOp(opName);
+  assert(opHandle && "could not find convolution op");
+  auto opTyped = opHandle->typed<at::Tensor(
+      const at::Tensor &, const at::Tensor &, const c10::optional<at::Tensor> &,
+      const at::IntArrayRef, const at::IntArrayRef, const at::IntArrayRef,
+      const bool, const at::IntArrayRef, const int64_t)>();
+
+  // Exclude recursive calls: convolution is completely emitted by this
+  // kernel.
+  c10::DispatchKeySet keySet{kAcapDispatchKey, kAcapGradDispatchKey};
+  c10::impl::ExcludeDispatchKeyGuard exclusion(keySet);
+
+  auto current = getCurrent();
```

Comment:
nit: out of context, getCurrent() with the auto was kind of hard for me to grok. if possible, avoiding auto or renaming this to getCurrentThreadAcapController() or some such.

(I came into this with an understanding of the opengl "makeCurrent" terminology; and was still confused until I literally looked up what getCurrent returns here; call it bad memory of what the "current thing" is in this situation :P )

Reply:
Renamed. The type is kind of gross.